src/engine/system/player_controlled.cpp

```diff
@@ -0,0 +1,31 @@
+#include "system.hpp"
+#include "engine/input/input_state.hpp"
+#include "engine/ecs.hpp"
+
+class SystemPlayerControlled: public System {
+  public:
+    void handle_components(ECS* ecs, InputState* input_state ) {
+      for (u32 ii=0; ii < ecs->comp_game_entity.size(); ii++) {
+        for (u32 jj=0; jj < ecs->comp_player_controlled.size(); ii++) {
+          if (ecs->comp_player_controlled[jj].entity_id == ecs->comp_game_entity[ii].entity_id) {
```

Comment:
Maybe do a 'reverse' if here - so something like `if (not the right ID) continue;`

Less indentation makes it nicer to read.